### ISSUE TITLE
出力の変更

### ISSUE THREAD

**Issue by Kalorie560** [Kalorie560/OptimalFert#17]
@claude

今回、提出ファイルの形式が id と Fertilizer Name の2つのカラムで構成されることが明確になりました（sample_submission.csv の通り）。この情報に基づき、特に以下の点について、開発プロセス全体で修正と再確認を徹底してください。

ターゲット変数の再定義:

train.csv 内の Fertilizer Name カラムを、予測すべきターゲット変数として扱ってください。
これは多クラス分類問題であることを意味します。Fertilizer Name に含まれるユニークな肥料名を特定し、それらを予測クラスとしてください。
モデルの出力調整:

選択する機械学習モデル（例: LightGBM, XGBoost, RandomForest, ニューラルネットワーク等）は、多クラス分類に対応したものを使用し、最終的に具体的な**肥料名（文字列）**を予測するように設定・調整してください。
ターゲット変数のエンコーディング（例: Label Encoding）と、予測後のデコーディング（数値ラベルから元の肥料名文字列への逆変換）の処理フローを確立してください。
評価指標の見直し:

これまでの評価指標の仮定（例：二値分類のROC AUC）から、多クラス分類に適した指標（例：Accuracy (正解率), Macro F1-score, LogLossなど）に焦点を切り替えてください。
ClearMLには、これらの多クラス分類指標を記録するようにしてください。
Submissionファイルの厳格なフォーマット遵守:

生成する submission.csv ファイルは、必ず id と Fertilizer Name の2つのカラムのみで構成されるようにしてください。
Fertilizer Name カラムには、test.csv の各 id に対応する予測された**肥料名（文字列）**を格納してください。sample_submission.csv の形式と完全に一致させてください。
ウェブアプリケーションの出力変更:

開発するウェブアプリケーションも、特徴量を入力した際に予測結果として**肥料名（文字列）**を表示するように修正してください。
ドキュメント (README) の更新:

日本語と英語のREADMEファイル内の「処理フロー」「モデル設計」「評価指標」などの関連セクションを更新し、このタスクが Fertilizer Name を予測する多クラス分類問題であることを正確に反映させてください。

**Comment by Kalorie560** [comment]
@claude
以下のエラーを修正
```
2025-06-05 05:51:45,220 - __main__ - INFO - All required packages are available ✅
2025-06-05 05:51:45,220 - __main__ - INFO - ============================================================
2025-06-05 05:51:45,220 - __main__ - INFO - STARTING OPTIMALFERT FERTILIZER PREDICTION PIPELINE
2025-06-05 05:51:45,220 - __main__ - INFO - ============================================================
2025-06-05 05:51:45,220 - __main__ - INFO - Directory 'data' ready
2025-06-05 05:51:45,220 - __main__ - INFO - Directory 'models' ready
2025-06-05 05:51:45,220 - __main__ - INFO - Directory 'outputs' ready
2025-06-05 05:51:45,220 - __main__ - INFO - Directory 'logs' ready
2025-06-05 05:51:45,220 - __main__ - INFO - Directory 'notebooks' ready
2025-06-05 05:51:45,223 - __main__ - INFO - 
========================================
2025-06-05 05:51:45,223 - __main__ - INFO - STEP 0: EXPERIMENT TRACKING SETUP
2025-06-05 05:51:45,223 - __main__ - INFO - ========================================
ClearML Task: created new task id=e4309aabd3924751b7e2f73c8cb576bc
2025-06-05 05:51:48.824027: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:477] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
E0000 00:00:1749102708.843339     703 cuda_dnn.cc:8310] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
E0000 00:00:1749102708.849361     703 cuda_blas.cc:1418] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2025-06-05 05:51:48.868673: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
ClearML results page: https://app.clear.ml/projects/8c93608c9aaa412893f21653a83f766b/experiments/e4309aabd3924751b7e2f73c8cb576bc/output/log
2025-06-05 05:51:53,967 - __main__ - INFO - ✅ ClearML experiment tracking initialized successfully
2025-06-05 05:51:53,968 - __main__ - INFO - Project: Playground-Series-S5E6
2025-06-05 05:51:53,971 - __main__ - INFO - Task URL: https://app.clear.ml/projects/8c93608c9aaa412893f21653a83f766b/experiments/e4309aabd3924751b7e2f73c8cb576bc/output/log
2025-06-05 05:51:53,972 - __main__ - INFO - 
========================================
2025-06-05 05:51:53,972 - __main__ - INFO - STEP 1: DATA PREPARATION
2025-06-05 05:51:53,973 - __main__ - INFO - ========================================
2025-06-05 05:51:53,973 - __main__ - INFO - Creating sample fertilizer data for demonstration...
2025-06-05 05:51:54,045 - src.data.preprocessing - INFO - Sample fertilizer data created in data/
2025-06-05 05:51:54,045 - src.data.preprocessing - INFO - Train shape: (1600, 14)
2025-06-05 05:51:54,046 - src.data.preprocessing - INFO - Test shape: (400, 13)
2025-06-05 05:51:54,046 - src.data.preprocessing - INFO - Unique fertilizers: 12
2025-06-05 05:51:54,047 - src.data.preprocessing - INFO - Fertilizer distribution in training data:
2025-06-05 05:51:54,051 - src.data.preprocessing - INFO - Fertilizer Name
NPK 20-20-20         599
DAP (18-46-0)        526
Urea (46-0-0)        205
MOP (0-0-60)         140
Organic Compost       45
Calcium Nitrate       33
NPK 15-15-15          29
Magnesium Sulfate      8
Bone Meal              6
NPK 10-10-10           5
Fish Emulsion          3
Potassium Sulfate      1
Name: count, dtype: int64
2025-06-05 05:51:54,053 - __main__ - INFO - 
========================================
2025-06-05 05:51:54,053 - __main__ - INFO - STEP 2: EXPLORATORY DATA ANALYSIS
2025-06-05 05:51:54,054 - __main__ - INFO - ========================================
2025-06-05 05:51:54,061 - src.visualization.eda - INFO - Loaded training data with shape: (1600, 14)
2025-06-05 05:51:54,063 - src.visualization.eda - INFO - Starting comprehensive EDA analysis...
2025-06-05 05:51:54,071 - src.visualization.eda - INFO - Dataset shape: (1600, 14)
2025-06-05 05:51:54,071 - src.visualization.eda - INFO - Target distribution: {'NPK 20-20-20': 599, 'DAP (18-46-0)': 526, 'Urea (46-0-0)': 205, 'MOP (0-0-60)': 140, 'Organic Compost': 45, 'Calcium Nitrate': 33, 'NPK 15-15-15': 29, 'Magnesium Sulfate': 8, 'Bone Meal': 6, 'NPK 10-10-10': 5, 'Fish Emulsion': 3, 'Potassium Sulfate': 1}
2025-06-05 05:51:54,072 - src.visualization.eda - INFO - Number of target classes: 12
2025-06-05 05:51:54,082 - src.visualization.eda - WARNING - Found missing values in 1 features
2025-06-05 05:51:54,168 - src.visualization.eda - INFO - Analyzed 9 numeric features
2025-06-05 05:51:54,171 - __main__ - ERROR - EDA failed: agg function failed [how->mean,dtype->object]
2025-06-05 05:51:54,171 - __main__ - ERROR - 
❌ Training pipeline failed!
```
